tron/bin/recover_batch.py

```diff
@@ -53,48 +54,41 @@ def notify(notify_queue, ignored, filepath, mask):
             # "A negative value -N indicates that the child was terminated by signal N (POSIX only)."
             # We should always exit with a positive code, so we take the absolute value of the return code
             exit_code = abs(return_code)
-            error_msg = (
-                'Action run killed by signal '
-                f'{signal.Signals(exit_code).name}'
-            )
+            error_msg = f'Action run killed by signal {signal.Signals(exit_code).name}'
         else:
             exit_code = return_code
-
-    elif pid is None or not psutil.pid_exists(pid):
+    elif pid is None:
+        log.warning(f"Status file {filepath.path} didn't have a PID. Will watch the file for updates.")
```

Comment:
I am a little afraid this will hang, but since @qui has suggested otherwise, I am willing to try it out.

Reply:
As discussed, we kind want it to hang by design